Common/util/string_utils.cpp

```diff
@@ -203,5 +204,12 @@ void StrUtil::WriteString(const String &s, Stream *out)
         out->Write(s.GetCStr(), len);
 }
 
+void StrUtil::WriteCStr(const String &s, Stream *out, int max_len)
+{
+    max_len = Math::Min(max_len, s.GetLength());
```

Comment:
Instead of silently truncating, would it be better to raise an error if it doesn't fix within max_len?